Integrations/src/main/java/io/deephaven/integrations/python/PythonTimeComponents.java

```diff
@@ -0,0 +1,118 @@
+package io.deephaven.integrations.python;
+
+import io.deephaven.util.annotations.ScriptApi;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.Period;
+import java.time.ZonedDateTime;
+
+/**
+ * Utility functions for {@code time.py} to extract {@link java.time} components in a single call.
+ */
+@ScriptApi
+public class PythonTimeComponents {
+
+    /**
+     * Extracts the components from a {@link LocalTime}. Equivalent to {@code new int[] {dt.getHour(), dt.getMinute(),
+     * dt.getSecond(), dt.getNano()}}.
+     *
+     * @param dt the local time
+     * @return the components
+     */
+    @ScriptApi
+    public static int[] getComponents(LocalTime dt) {
```

Comment:
The name `getComponents` is fine, but not as explicit as `getHMSN`. More of an observation, not pushing for change.